### PR TITLE
Add basic telemetry

### PR DESCRIPTION
**Pull request by AlekSi**
## Readiness checklist

* [ ] I added tests for new functionality or bugfixes.
* [x] I ran `task all`, and it passed.
* [x] I added/updated comments for both exported and unexported top-level declarations (functions, types, etc).
* [ ] I checked comments rendering with `task godocs`.
* [x] I ensured that the title is good enough for the changelog.
* [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Assignee, Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -89,7 +89,10 @@ linters-settings:
   revive:
     ignore-generated-header: true
     severity: warning
-    enableAllRules: true
```
Comment (AlekSi) [inline .golangci.yml:92]: That wasn't a correct spelling

**Review by chilagrow — DISMISSED** [review]
looks good!

File: internal/util/telemetry/telemetry.go [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+// Run runs reporter until context is canceled.
+func (r *Reporter) Run(ctx context.Context) {
+	r.L.Debug("Reporter started.")
+	defer r.L.Debug("Reporter stopped.")
+
+	ch := r.P.Subscribe()
+
+	r.firstReportDelay(ctx, ch)
+
+	for ctx.Err() == nil {
```
[the remark this answers is not in the document]
Comment (AlekSi) [reply internal/util/telemetry/telemetry.go:85]: That context is cancelled only when the whole FerretDB is being stopped

File: internal/util/telemetry/telemetry.go [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+type NewReporterOpts struct {
+	URL            string
+	P              *state.Provider
+	L              *zap.Logger
+	UndecidedDelay time.Duration
+	ReportInterval time.Duration
+	ReportTimeout  time.Duration
+}
+
+// NewReporter create a new reporter.
```
Comment (rumyantseva) [inline internal/util/telemetry/telemetry.go:68]:
```suggestion
// NewReporter creates a new reporter.
```

File: internal/util/telemetry/telemetry.go [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+	"github.com/FerretDB/FerretDB/internal/util/state"
+	"github.com/FerretDB/FerretDB/internal/util/version"
+)
+
+// request represents telemetry request.
+type request struct {
+	Version string `json:"version"`
+	Commit  string `json:"commit"`
+	Branch  string `json:"branch"`
+	Dirty   bool   `json:"dirty"`
```
Comment (rumyantseva) [inline internal/util/telemetry/telemetry.go:38]: What is `Dirty` for?

File: internal/util/telemetry/telemetry.go [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+	URL            string
+	P              *state.Provider
+	L              *zap.Logger
+	UndecidedDelay time.Duration
+	ReportInterval time.Duration
+	ReportTimeout  time.Duration
+}
+
+// NewReporter create a new reporter.
+func NewReporter(opts *NewReporterOpts) (*Reporter, error) {
```
Comment (rumyantseva) [inline internal/util/telemetry/telemetry.go:69]: Do we really need `error` in the signature? Do you think we might need it later?

File: cmd/ferretdb/main.go [hunk 6]
```diff
@@ -69,6 +76,12 @@ var cli struct {
 
 	Test struct {
 		RecordsDir string `default:""  help:"Testing flag: directory for record files."`
+		Telemetry  struct {
+			URL            string        `default:"https://beacon.ferretdb.io/" help:"Testing flag: telemetry URL."`
```
Comment (rumyantseva) [inline cmd/ferretdb/main.go:80]: How would it work when we run FerretDB locally for dev/testing purposes? It seems that by default all the telemetry is sent to `beacon.ferretdb.io`. Is it on purpose? Will we be able to sort out dev instances etc?

**Review by rumyantseva — COMMENTED** [review]
LGTM to me overall. I asked a few questions.

File: internal/util/telemetry/telemetry.go (hunk 4 above)
Comment (AlekSi) [reply internal/util/telemetry/telemetry.go:38]: To check if someone encounters weird problem with non-clean build with modified source. In that case our response could be "please try an official build"

File: internal/util/telemetry/telemetry.go (hunk 5 above)
Comment (AlekSi) [reply internal/util/telemetry/telemetry.go:69]: I think the next PR will need it. If not, I will remove it

File: cmd/ferretdb/main.go (hunk 6 above)
Comment (AlekSi) [reply cmd/ferretdb/main.go:80]: I don't think that really matters. We will be able to exclude our own instances just like we would exclude other testing instances – by small uptime.

**Review by rumyantseva — APPROVED** [review]
No new questions. LGTM

**Comment by AlekSi** [comment]
@chilagrow PTAL

**Pull request merged**